File: Application/front/lib/models/item.dart
import 'package:front/models/item_type.dart';

class Item {
  final String id;
  final String label;
  final ItemType type;
  final String? description;
  final Target? target;
  final int? quantity;
  int ratio;
  int price;
  int duration;

  Item(this.id, this.label, this.type, this.description, this.target,
      this.ratio, this.price, this.duration, this.quantity);

  static Item fromJSON(data) {
    ItemType type = ItemTypeParsing.fromString(data['type'])!;
    Target? target = TargetParsing.fromString(data['target'] ?? "");
    num ratio = data['ratio'];
    num price = data['price'];
    num duration = data['duration'];
    num? quantity = data['quantity'];

    return Item(
        data['_id'].toString(),
        data['label'],
        type,
        data['description'],
        target,
        ratio.toInt(),
        price.toInt(),
        duration.toInt(),
        quantity?.toInt());
  }
}

File: Application/game-engine/models/data/item.js
export class Item{
    initFromObj(o){
        this.label = o.label;
        this.type = o.type;
        this.desc = o.desc;
        this.target = o.target;
        this.ratio = o.ratio;
        this.price = o.price;
        this._id = o._id;
        this.duration = o.duration;
        this.quantity = 0;
    }
    get Price(){
        return this.price;
    }
    set Price(price){
        this.price = price;
    }
    get Label(){
        return this.label;
    }
    get Type(){
        return this.type;
    }
    get Desc(){
        return this.desc
    }
    get Target(){
        return this.target;
    }
    get Ratio(){
        return this.ratio;
    }
    get Id(){
        return this._id;
    }
    set Id(id){
        this._id = id;
    }
    get Duration(){
        return this.duration;
    }
    get Quantity(){
        return this.quantity;
    }
    set Quantity(quantity){
        this.quantity = quantity;
    }

}
